source/server/admin/grouped_stats_request.cc

```diff
@@ -0,0 +1,156 @@
+#include "source/server/admin/grouped_stats_request.h"
+
+#include <string>
+#include <vector>
+
+#include "source/server/admin/prometheus_stats_formatter.h"
+#include "source/server/admin/stats_params.h"
+#include "source/server/admin/stats_render.h"
+
+namespace Envoy {
+namespace Server {
+
+GroupedStatsRequest::GroupedStatsRequest(Stats::Store& stats, const StatsParams& params,
+                                         Stats::CustomStatNamespaces& custom_namespaces,
+                                         UrlHandlerFn url_handler_fn)
+    : StatsRequest(stats, params, url_handler_fn), custom_namespaces_(custom_namespaces),
+      global_symbol_table_(stats.constSymbolTable()) {
+
+  // the "type" query param is ignored for prometheus stats, so always start from
```

Comment:
Hm.

I suggested renaming this GroupedStatsRequest and I'm fine with sticking with it, but right now there's prometheus-specific stuff in here -- type 'type' policy, the prometheus-specific text-readouts policy, maybe other things.

Maybe just leave a TODO to clean this up if we decide to build a grouped view in HTML or JSON.

Reply:
I have included a TODO in the header file.